extensions/wikia/Recirculation/services/FandomDataService.class.php

```diff
@@ -97,15 +97,17 @@ private function dedupePosts( $rawPosts ) {
 		$posts = [];
 		$postIds = [];
 
-		foreach ( $rawPosts as $post ) {
-			if ( count( $posts ) >= self::PARSELY_POSTS_LIMIT ) {
-				break;
-			}
-
-			$metadata = json_decode( $post->metadata );
-			if ( !empty( $metadata->postID ) && !in_array( $metadata->postID, $postIds ) ) {
-				$postIds[] = $metadata->postID;
-				$posts[] = $post;
+		if ( is_array( $rawPosts ) ) {
```

Comment:
My preference would be an early return:
    if ( !is_array( $rawPosts) ) {
        return [];
    }
Also I would rather make sure the `$rawPosts` passed are always an array (validate before calling this method), unless there's a reason to do it in here and not before.